components/net/filemanager_thread.rs

```diff
@@ -309,8 +310,14 @@ impl <UI: 'static + UIProvider> FileManagerStore<UI> {
 
     fn select_file(&self, patterns: Vec<FilterPattern>,
                    sender: IpcSender<FileManagerResult<SelectedFile>>,
-                   origin: FileOrigin) {
-        match self.ui.open_file_dialog("", patterns) {
+                   origin: FileOrigin, opt_test_path: Option<String>) {
+        let opt_s = if select_files_pref_enabled() {
+            opt_test_path
+        } else {
+            self.ui.open_file_dialog("", patterns)
+        };
```

Comment:
Yep. Leave a comment here though mentioning that while this pref check is redundant, it's in place to ensure that a compromised script thread cannot request arbitrary files. r=me